src/main/jbake/content/blog/2017/01/13/events-aggregation-extension.adoc

```diff
@@ -0,0 +1,167 @@
+= Extending Complex Event Processing in Hawkular Alerting
+Lucas Ponce
+2017-01-13
+:jbake-type: post
+:jbake-status: published
+:jbake-tags: blog, alerts, complex, events, processing, stream, polling
+
+== Stream Processing versus Polling Processing
+
+Hawkular Alerting uses different techniques to detect behaviours by defining rules.
+
+The main implementation is based in a Stream Processing design. +
+Hawkular Alerting analyzes streams of data and events ordered in a timeline searching matches against conditions. +
+This technique is also known as Event Processing (or Complex Event Processing)specially when the processing of incoming data might need multiple conditions.
+
+This method is particularity good to identify meaningful happenings on complex scenarios and respond to them as quickly as possible.
+
+Hawkular Alerting also supports Polling techniques based on Alerters that allow to define queries that are processing
+against a backend in a periodically manner.
+
+This Polling technique is also good in non real time scenarios, for example, to compare historical values in
+large range time intervals using statistical operators like averages, medians or percentiles.
+
+== Sliding Windows on Complex Event Processing
+
+An interesting aspect in the logic of detection behaviours is when we can combine characteristics of Stream Processing and Polling Processing.
+
+A Sliding Window is a way to define a scope in the stream of data and events received in the timeline. +
+This Sliding Window let us to define special rules that apply only on the scoped data and events. +
+
+A powerful use case is when we can aggregate the scoped events and define expressions on them. +
+For example, let's examine the high level scenarios
+
+    - *Marketing* +
+    _Detect when a customer buys several items in a short period of time_
+
+    - *Fraud* +
+    _Alert when a customer buys from several locations in a short period of time_
+
+    - *Customer loyalty* +
+    _Detect specific transactions to offer premium discounts to customers_
+
+All of these scenarios need some sort of aggregation (i.e. aggregate events by customer/transaction id),
+defines expressions on aggregated fields and scope these events using a Slide Window.
+
+== EventsAggregation Extension
+
+Hawkular Alerting has introduced a new Extension called EventsAggregation. +
+Extensions is a new mechanism to pre-process Data or Events before they are processing by the Alerting engine.
```

Comment:
You might want to put "Extensions" in italics or something (quotes?), as it is being used as a name of a mechanism, but is plural.
"processed" instead of "processing"